compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -194,6 +194,12 @@ trait QuotesAndSplices {
         using spliceContext.retractMode(Mode.QuotedPattern).withOwner(spliceOwner(ctx)))
     pat.select(tpnme.Underlying)
 
+  def typedHole(tree: untpd.Hole, pt: Type)(using Context): Tree = {
+    val tpt = typedType(tree.tpt)
+    // TODO check args and contents
```

Comment:
Would be good to expand the TODO a bit on what needs to be checked and whether this is needed for soundness or for detecting bugs in the compiler.

Reply:
Done.